### ISSUE TITLE
[Craft 4]: Supertables inside matrix blocks don't update after save

### ISSUE THREAD

**Issue by Antimated** [engram-design/SuperTable#473]
**Description**
I have a matrix content builder with supertables inside some of the blocks. If I update the contents of a field that lives inside a supertable (e.g. change the handle, change the name, ...) and then save my matrix, the fields are not updated. Also no YAML changes are being detected.

**Steps to reproduce**
1. Create a matrix with a few matrix blocks
2. Inside those matrix block place a supertable (I use a NON static supertable with matrix layout as it's field layout)
3. Inside the supertable you add a plain text field and a redactor field.
4. Change one of the fields field settings (name, handle, redactor config, ...)
5. Save
6. Observe

**Additional info**
- Plugin version: 3.0.0-beta.4
- Craft version: Craft Pro 4.0.4
- Redactor version: 3.0.0
- Multi-site: Yes

**Additional context**
Seems like my colleague doesn't have this problem on the same install but I do. Both running locally on the same env (laravel valet, but we both have our separate db's)

EDIT: seems to be only affected by supertables because regular fields inside other matrix blocks save without issue :/

**Comment by bymayo** [comment]
Yeah we're also having this issue with the same versions.

My issue was I could create the field inside the SuperTable, but going back to edit the label/handle just wouldn't save. But give me a success message asif it had done it.

**Comment by iamtompickering** [comment]
Yeah just come across the same issue. 

I tried adding a new field to an existing Super Table field in a Matrix field and it just won't save the new field. Nothing in project YAML is updated either. 

Craft version: 4.0.4
Plugin version: 3.0.0-beta.4

**Comment by tim-digitalastronaut** [comment]
We're also encountering the very same issue. Same versions.

**Comment by engram-design** [comment]
Can you try the latest `craft-4` branch with a fix for this? Change your `verbb/super-table` requirement in `composer.json` to:

```
"require": {
  "verbb/super-table": "dev-craft-4 as 3.0.0-beta.4",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
This should be fixed in [3.0.0-beta.5](https://github.com/verbb/super-table/releases/tag/3.0.0-beta.5)

**Comment by mediagirl** [comment]
This update got saving working correctly for me again. Thank you for quick fix!

**Comment by Antimated** [comment]
@engram-design Maybe on another note but kinda related: Supertable blocks inside a matrix don't remember their order after saving. Running 3.0.0-beta.5 of the plugin :)

**Comment by mmikkel** [comment]
@Antimated That's probably the same issue as https://github.com/verbb/super-table/issues/468 (fixed but no tagged release available yet)

**Comment by engram-design** [comment]
@Antimated Fixed in [3.0.0-beta.6](https://github.com/verbb/super-table/releases/tag/3.0.0-beta.6)

**Comment by Antimated** [comment]
@engram-design thanks! looks to work as expected now :)!
